lib/kitchen/driver/oci.rb

```diff
@@ -100,23 +83,51 @@ def destroy(state)
         return unless state[:server_id]
 
         instance.transport.connection(state).close
-        comp_api.terminate_instance(state[:server_id])
+
+        if instance_type == 'compute'
+          comp_api.terminate_instance(state[:server_id])
+        elsif instance_type == 'dbaas'
+          dbaas_api.terminate_db_system(state[:server_id])
+        end
 
         state.delete(:server_id)
         state.delete(:hostname)
       end
 
+      def process_freeform_tags(freeform_tags)
+        prov = instance.provisioner.instance_variable_get(:@config)
+        tags = %w[run_list policyfile]
+        tags.each do |tag|
+          freeform_tags[tag] = prov[tag.to_sym].join(',') unless prov[tag.to_sym].nil? || prov[tag.to_sym].empty?
+        end
+        freeform_tags[:kitchen] = true
+        freeform_tags
+      end
+
+      def process_windows_options(state)
+        state[:username] = config[:winrm_user] if config[:setup_winrm]
+        if config[:setup_winrm] == true &&
+           config[:password].nil? &&
+           state[:password].nil?
+          state[:password] = config[:winrm_password] || random_password
+        end
+        state
+      end
+
       private
 
+      def instance_type
+        config[:instance_type].downcase
```

Comment:
Maybe this is a more sensible place to put a general check for valid values of `config[:instance_type]` (and raise an exception here) so that we don't have to keep checking for it everywhere else as I suggested in my other comments.